### PR TITLE
add cronjob to run the ndmis performance report job every morning

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

add cronjob to run the ndmis performance report job every morning

## What is the intent behind these changes?

run the report using automated cronjob

### CODE REVIEW

**Comment by tomsmyers** [comment]
PLEASE DO NOT MERGE

**Comment by sldblog** [comment]
> PLEASE DO NOT MERGE

We could, if it's behind a `{{- if not .Values.env_details.production_env -}}`

File: helm_deploy/hmpps-interventions-service/templates/cronjob-ndmis-performance-report.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: generate-ndmis-performance-report
+spec:
+  schedule: "0 3 * * *"
```
[the remark this answers is not in the document]
Comment (tomsmyers) [reply helm_deploy/hmpps-interventions-service/templates/cronjob-ndmis-performance-report.yaml:6]: 👍

File: helm_deploy/hmpps-interventions-service/templates/cronjob-ndmis-performance-report.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          containers:
+            - name: hmpps-interventions-service
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+              imagePullPolicy: Always
+              args: ["--jobName=ndmisPerformanceReportJob"]
+  {{- include "deployment.envs" . | nindent 14 }}
+                # this is a bit of a failsafe - it shouldn't happen, but
+                # we don't want this job running migrations on startup
+                - name: SPRING_FLYWAY_ENABLED
+                  value: "false"
```
[the remark this answers is not in the document]
Comment (tomsmyers) [reply helm_deploy/hmpps-interventions-service/templates/cronjob-ndmis-performance-report.yaml:21]: 👍

**Review by sldblog — APPROVED** [review]
LGTM 👍

**Comment by sldblog** [comment]
Tested with
`helm template --set image.tag=my.fake.release hmpps-interventions-service hmpps-interventions-service --values=values-preprod.yaml | yq e -`

**Pull request merged**